`azurerm_web_pubsub_hub` - add `event_listener` property (#21145)
hashicorp/terraform-provider-azurerm: website/docs/r/web_pubsub_hub.html.markdown

---

An `event_listener` block supports the following:

* `system_event_name_filter` - (Optional) Specify the list of system events. Supported values are `connect`, `connected` and `disconnected`.

> * `user_event_name_filter` - (Optional) Specify the matching event names. There are 3 kind of patterns supported: * `*` matches any event name * `,` Combine multiple events with `,` for example `event1,event2`, it matches event `event1` and `event2` * The single event name, for example `event1`, it matches `event1`. Defaults to `"*"`

Reviewer: shouldn't this property be a lint then instead of CSV?

Author: Thanks kt for the review, sorry that I can't quite follow your question, can you be more specific about it? Thanks

Reviewer: sorry list. ie as it is `*` or `event1,event2` it could be user_event_name_filter = ["*"] and user_event_name_filter = ['event1","event2"]

-> **NOTE:** User can change the order of `event_handler` to change the priority accordingly.

* `event_listener` - (Optional) An `event_listener` block as defined below.

> -> **NOTE:**  The managed identity of Web PubSub service must be enabled, and the identity should have the "Azure Event Hubs Data sender" role to access Event Hub.

Reviewer 2: ```suggestion
-> **NOTE:**  The managed identity of Web PubSub service must be enabled and the identity must have the "Azure Event Hubs Data sender" role to access the Event Hub.
```

* `user_event_pattern` - (Optional) Specify the matching event names. There are 3 kind of patterns supported: * `*` matches any event name * `,` Combine multiple events with `,` for example `event1,event2`, it matches event `event1` and `event2` * The single event name, for example `event1`, it matches `event1`.

> * `system_events` - (Optional) Specify the list of system events. Supported values are `connected` and `disconnected`.

Reviewer 2: According to schema this appears to support 3 values
```suggestion
* `system_events` - (Optional) Specifies the list of system events. Supported values are `connect`, `connected` and `disconnected`.
```

* `auth` - (Optional) An `auth` block as defined below.

---

> An `event_listener` block supports the following:
> 
> * `system_event_name_filter` - (Optional) Specify the list of system events. Supported values are `connected` and `disconnected`.
> 
> * `user_event_name_filter` - (Optional) Specify the list of matching user event names. If the filter is to match all events, use `["*"]`.
>  
> * `eventhub_namespace_name` - (Required) Specify the event hub namespace name to receive the events.
> 
> * `eventhub_name` - (Required) Specify the event hub name to receive the events.
> 

Reviewer 2: ```suggestion
An `event_listener` block supports the following:

* `system_event_name_filter` - (Optional) Specifies a list of system events. Supported values are `connected` and `disconnected`.

* `user_event_name_filter` - (Optional) Specifies a list of matching user event names. `["*"]` can be used to match all events.
 
* `eventhub_id` - (Required) Specifies the Event Hub that will receive the events.

```
